### PR TITLE
Fix equating comparisons between MPANs

### PR DESCRIPTION
**Pull request by limedaniel**
> *I'm undecided as to whether this should be a bump from `1.0.2` to `1.0.3` or to `1.1.0` on account of the API sort-of-change.*

I noticed today that `MPAN(string) ≠ MPAN(string)` on account of the fact that I never implemented `__eq__()` on the `MPAN` class.  This PR fixes that.

### CODE REVIEW

File: tests/test_mpan.py [hunk 1]
```diff
@@ -156,6 +156,20 @@ def test___repr__(self):
...
+
+    def test___eq__with_invalid_mpans(self):
+        for string in INVALID:
+            self.assertEqual(MPAN(string), MPAN(string))
+
+    def test___eq__with_foreign_objects(self):
+        mpan_string = "108011002099999999386"
+        self.assertNotEqual(MPAN(mpan_string), mpan_string)
+        self.assertNotEqual(MPAN(mpan_string), 108011002099999999386)
+        self.assertNotEqual(MPAN(mpan_string), "twelve")
```
Comment (roblevy) [inline tests/test_mpan.py:171]: Ha ha.

**Review by roblevy — APPROVED** [review]
Looks good. Why are we keeping a list of bugfixes in the README? Is that standard practice?

**Comment by limedaniel** [comment]
It's a good practise to have a changelog, and for smallish projects, it's common to have the changelog in the readme, especially since it means that the changelog appears on PyPI.

Before I go ahead and merge this, what're your thoughts on:

```python
MPAN(mpan_str) ≠ mpan_str ?
MPAN(mpan_int) ≠ mpan_int ?
```

and

Should this be a patch bump (1.0.3) or minor bump (1.1.0)?

File: mpan.py [hunk 2]
```diff
@@ -365,6 +365,11 @@ def __str__(self) -> str:
     def __repr__(self):
         return str(self)
 
+    def __eq__(self, other):
+        if not isinstance(other, MPAN):
+            return False
+        return str(self) == str(other)
```
Comment (remy-naylor) [inline mpan.py:371]: Yep, this is exactly the logic I'd expect. I don't care for JSesque `MPAN('blah') == 'blah'` being `True`!

**Review by remy-naylor — APPROVED** [review]

**Comment by remy-naylor** [comment]
> Should this be a patch bump (1.0.3) or minor bump (1.1.0)?

Minor seems fine to me, this is the sort of thing you would reasonably expect to be implemented already, so it doesn't feel like a new feature per se.

**Pull request merged**